### PR TITLE
Add GMCreateGold

### PR DESCRIPTION
**Pull request by sarimash**
### All Submissions
image

* [x] Have you followed the guidelines in our Contributing document?
* [x] Can you post a screenshot of your changes (if applicable)?

### New Feature Submissions

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?

### Changes to Core Features

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?

### CODE REVIEW

File: server/src/helpers/game/commands/gm/GMCreateGold.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  canBeInstant = false;
+  canBeFast = false;
+
+  execute(player: IPlayer, args: IMacroCommandArgs) {
+    const gold = this.game.itemCreator.getSimpleItem('Gold Coin');
+    const value = +args.arrayArgs[0];
+    if (!value) {
+      this.sendMessage(player, `Enter an amount of gold.`);
+      return;
+    } else {
```
Comment (seiyria) [inline server/src/helpers/game/commands/gm/GMCreateGold.ts:17]: The `else` is superfluous if you do `if() { return }` - basically pull this code out of the else block.

File: server/src/helpers/game/commands/gm/GMCreateGold.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  aliases = ['@gold'];
+  isGMCommand = true;
+  canBeInstant = false;
+  canBeFast = false;
+
+  execute(player: IPlayer, args: IMacroCommandArgs) {
+    const gold = this.game.itemCreator.getSimpleItem('Gold Coin');
+    const value = +args.arrayArgs[0];
+    if (!value) {
```
Comment (seiyria) [inline server/src/helpers/game/commands/gm/GMCreateGold.ts:14]: This needs to check if(!value || isNaN(value))

File: server/src/helpers/game/commands/gm/GMCreateGold.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+export class GMCreateGold extends MacroCommand {
+
+  aliases = ['@gold'];
+  isGMCommand = true;
+  canBeInstant = false;
+  canBeFast = false;
+
+  execute(player: IPlayer, args: IMacroCommandArgs) {
+    const gold = this.game.itemCreator.getSimpleItem('Gold Coin');
+    const value = +args.arrayArgs[0];
```
Comment (seiyria) [inline server/src/helpers/game/commands/gm/GMCreateGold.ts:13]: This needs to do Math.round() so people can't make 1.1 gold.

File: server/src/helpers/game/commands/gm/GMCreateGold.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    if (!value) {
+      this.sendMessage(player, `Enter an amount of gold.`);
+      return;
+    } else {
+      gold.mods.value = value;
+      this.game.worldManager.getMap(player.map).state.addItemToGround(player.x, player.y, gold);
```
Comment (seiyria) [inline server/src/helpers/game/commands/gm/GMCreateGold.ts:19]: It should *probably* add it to your right hand if that's empty, first. If the right hand is empty, then drop it on the ground. You can see an example of this with the GM Item command.

File: server/src/helpers/game/commands/gm/GMCreateGold.ts (hunk 1 above)
Comment (sarimash) [reply server/src/helpers/game/commands/gm/GMCreateGold.ts]: Thanks, done

File: server/src/helpers/game/commands/gm/GMCreateGold.ts (hunk 2 above)
Comment (sarimash) [reply server/src/helpers/game/commands/gm/GMCreateGold.ts]: Thanks, added this and a few other checks. There's still a cute interaction where a hex value is a valid number, but that seems fine I believe

File: server/src/helpers/game/commands/gm/GMCreateGold.ts (hunk 3 above)
Comment (sarimash) [reply server/src/helpers/game/commands/gm/GMCreateGold.ts]: Thanks, did that as well as taking the absolute value of the entered number

File: server/src/helpers/game/commands/gm/GMCreateGold.ts (hunk 4 above)
Comment (sarimash) [reply server/src/helpers/game/commands/gm/GMCreateGold.ts:24]: Thanks, done. It ended up having another else block similar to the old one, but this one doesn't have any returns so I assume we actually want this block

**Pull request merged**